### PR TITLE
Fix UI :  Redirect issue after token expiry user login

### PR DESCRIPTION
**Pull request by Ashish8689**
### Describe your changes :

-  fix redirect issue after token expiry of user and when they try to re-login.

#
### Type of change :

- [x] Bug fix

#
### Frontend Preview (Screenshots) :

https://user-images.githubusercontent.com/66266464/191533347-8d6e855f-ee9c-4b58-b64b-9413456be5d6.mov

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#
### Reviewers
@open-metadata/ui

### CODE REVIEW

File: openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx [hunk 1]
```diff
@@ -122,13 +122,7 @@ export const AuthProvider = ({
...
   const handledVerifiedUser = () => {
     if (!isProtectedRoute(location.pathname)) {
-      const urlPathname = cookieStorage.getItem(REDIRECT_PATHNAME);
-      if (urlPathname) {
-        cookieStorage.removeItem(REDIRECT_PATHNAME);
-        history.push(urlPathname);
-      } else {
-        history.push(ROUTES.HOME);
-      }
+      history.push(ROUTES.HOME);
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx:125]: @Ashish8689 , Can you confirm what we are trying to do here?

File: openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx [hunk 2]
```diff
@@ -74,13 +80,22 @@ const PermissionProvider: FC<PermissionProviderProps> = ({ children }) => {
     return AppState.getCurrentUserDetails();
   }, [AppState.userDetails, AppState.nonSecureUserDetails]);
 
+  const handledVerifiedUser = () => {
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx:83]: Here method name is `handledVerifiedUser` but I can not see any user-related stuff. Can you give an appropriate name to the method?

File: openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx [hunk 3]
```diff
@@ -267,6 +265,7 @@ export const AuthProvider = ({
    */
   const trySilentSignIn = () => {
     const pathName = location.pathname;
+    storeRedirectPath(pathName);
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx:268]: Why do we need this here? can you please explain?

File: openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx [hunk 4]
```diff
@@ -74,13 +80,22 @@ const PermissionProvider: FC<PermissionProviderProps> = ({ children }) => {
...
+  };
+
   /**
    * Fetch permission for logged in user
    */
   const fetchLoggedInUserPermissions = async () => {
     try {
       const response = await getLoggedInUserPermissions();
       setPermissions(getUIPermission(response.data || []));
+      handledVerifiedUser();
```
Comment (Sachin-chaurasiya) [inline openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx:98]: Do we need this in the permission provider?

File: openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx (hunk 1 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/authentication/auth-provider/AuthProvider.tsx:125]: @Sachin-chaurasiya After successful login user will be redirected to home page.

File: openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx (hunk 2 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx:83]: Updating!

File: openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx (hunk 4 above)
Comment (Ashish8689) [reply openmetadata-ui/src/main/resources/ui/src/components/PermissionProvider/PermissionProvider.tsx:98]: @Sachin-chaurasiya Yes, as for redirecting user in a Protected route `LoggedInUserPermissions` should be present. If not, user will redirect user to 404 after login which is not expected.

**Review by Sachin-chaurasiya — APPROVED** [review]

**Pull request merged**